backend/src/exercise/exercise-wrapper.ts

```diff
@@ -1,14 +1,161 @@
-import type { ExerciseAction, Role } from 'digital-fuesim-manv-shared';
-import { reduceExerciseState, ExerciseState } from 'digital-fuesim-manv-shared';
-import type { ActionEmitter } from './action-wrapper';
+import type { ExerciseAction, Role, UUID } from 'digital-fuesim-manv-shared';
+import {
+    ExerciseState,
+    reduceExerciseState,
+    validateExerciseAction,
+} from 'digital-fuesim-manv-shared';
+import type { EntityManager } from 'typeorm';
+import { IncrementIdGenerator } from '../utils/increment-id-generator';
+import { ValidationErrorWrapper } from '../utils/validation-error-wrapper';
+import { ExerciseWrapperEntity } from '../database/entities/exercise-wrapper.entity';
+import { NormalType } from '../database/normal-type';
+import type { DatabaseService } from '../database/services/database-service';
+import { Config } from '../config';
 import { ActionWrapper } from './action-wrapper';
 import type { ClientWrapper } from './client-wrapper';
 import { exerciseMap } from './exercise-map';
 import { patientTick } from './patient-ticking';
 import { PeriodicEventHandler } from './periodic-events/periodic-event-handler';
 
-export class ExerciseWrapper {
-    private tickCounter = 0;
+export class ExerciseWrapper extends NormalType<
+    ExerciseWrapper,
+    ExerciseWrapperEntity
+> {
+    private _changedSinceSave = true;
+
+    public get changedSinceSave() {
+        return this._changedSinceSave;
+    }
+
+    /**
+     * Mark this exercise as being up-to-date in the database
+     */
+    public hasBeenSaved() {
```

Comment:
Rename to `markSaved()`. Currently, I'd expect this function to return a boolean indicating whether it has been saved